crypto/s2n_hmac.c

```diff
@@ -80,75 +80,38 @@ int s2n_hmac_is_available(s2n_hmac_algorithm hmac_alg)
 
 static int s2n_sslv3_mac_init(struct s2n_hmac_state *state, s2n_hmac_algorithm alg, const void *key, uint32_t klen)
 {
-    if (!s2n_hmac_is_available(alg)) {
```

Comment:
This check prevents us from using `S2N_HMAC_MD5`, `S2N_HMAC_SSLv3_MD5`, and `S2N_HMAC_SSLv3_SHA1` when in FIPS mode. It's been removed here, but hasn't been added back anywhere else.

Reply:
The check is actually still there, on line 175 in the new code (github folded this line away in the diff).
https://github.com/awslabs/s2n/blob/feaf021270fbaf9f5b3ee96e7669c09642952444/crypto/s2n_hmac.c#L175

Since s2n_sslv3_mac_init is only called by s2n_hmac_init, leaving the check in s2n_sslv3_mac_init is redundant.